### DISCUSSION TITLE
I request to obtain the data from the database.

### DISCUSSION THREAD

**Discussion by AzielCF** [capacitor-community/sqlite#396] [Q&A] [answered]
Hello, I hope you are doing well.

I have a very specific question and it is that I have found this library, I have seen the documentation and examples and I feel completely confused.

Most of the examples I can see are for creating a SQLite database, which I have been able to achieve, but I cannot find a way or understandable and clear information to be able to read and obtain the data from this created database.

In this case, I use capacitor with an unencrypted database. This database is in the root directory "data/data/{appName}".
I have tried using a syntax type with createConnection() but this only lets me know that it is used to create a new database since when I run it more than once, it leaves a message that the previously created database already exists.

I am requesting help to understand this library.

Certainly, my goal is to be able to handle it like with SQL queries.
I am working with **Vue 3**.

**Comment by jepiqueau** [comment]
@AzielCF thanks for looking at the plugin. There are plenty of applications demonstrating the use of the plugin  in vue3 framework listed in the Readme.md file. I assume you find your way as you close the issue

**Reply by AzielCF** [reply]
I appreciate your response. I have been reviewing the repository and still find it confusing, although I have managed to make it work in Capacitor as I wanted.

Now I am trying to make it work in Electron as well, but I still do not understand how to do it. I am using the Quasar framework of Vue for multi-platform applications, and the codes for Capacitor and Electron are separated.

I am concerned that I will have to install the Capacitor CLI and the root folder for it to work with Electron. If you have any references that could help me understand better, it would be helpful, as I want to save the database in the AppData folder of the application so that I can connect to it and manipulate it later.

I don't know if this is possible, but in your examples, I have seen declarations for Electron to save the database in Windows, like this:

[https://github.com/jepiqueau/vite-vue-sqlite-app/blob/main/capacitor.config.ts](url)

I would appreciate it if you could help me with this :)

**Comment by jepiqueau** [comment, marked as answer]
@AzielCF I do not know the Quasar framework bur the procedure should be the same than for Vue.
make sure that your `capacitor.config.ts` includes the location of the databases for Electron
```
  plugins: {
    CapacitorSQLite: {
      iosDatabaseLocation: 'Library/CapacitorDatabase',
      iosIsEncryption: false,
      iosKeychainPrefix: appName,
      iosBiometric: {
        biometricAuth: false,
        biometricTitle : "Biometric login for capacitor sqlite"
      },
      androidIsEncryption: false,
      androidBiometric: {
        biometricAuth : false,
        biometricTitle : "Biometric login for capacitor sqlite",
        biometricSubTitle : "Log in using your biometric"
      },
      electronWindowsLocation: "C:\\ProgramData\\CapacitorDatabases",
      electronMacLocation: "/Volumes/Development_Lacie/Development/CapacitorDatabases",
      electronLinuxLocation: "Databases"
    }
  }
```

and the run the following commands

``` 
npm i --save @capacitor-community/electron@4.1.2
npm run build
npx cap sync @capacitor-community/electron
npx cap copy @capacitor-community/electron
cd electron
npm i --save jszip
npm i --save node-fetch@2.6.7
npm i --save sqlite3
npm i --save-dev @types/sqlite3
npm i --save electron-rebuild
npm run build
npm run electron-rebuild
npm run electron:start
```

This must start the electron application
